accountsdb-plugin-postgres/src/postgres_client/postgres_client_transaction.rs

```diff
@@ -0,0 +1,1335 @@
+/// Module responsible for handling persisting transaction data to the PostgreSQL
+/// database.
+use {
+    crate::{
+        accountsdb_plugin_postgres::{
+            AccountsDbPluginPostgresConfig, AccountsDbPluginPostgresError,
+        },
+        postgres_client::{DbWorkItem, ParallelPostgresClient, SimplePostgresClient},
+    },
+    chrono::Utc,
+    log::*,
+    postgres::{Client, Statement},
+    postgres_types::ToSql,
+    solana_accountsdb_plugin_interface::accountsdb_plugin_interface::{
+        AccountsDbPluginError, ReplicaTransactionInfo,
+    },
+    solana_runtime::bank::RewardType,
+    solana_sdk::{
+        instruction::CompiledInstruction,
+        message::{
+            v0::{self, AddressMapIndexes},
+            MappedAddresses, MappedMessage, Message, MessageHeader, SanitizedMessage,
+        },
+        transaction::TransactionError,
+    },
+    solana_transaction_status::{
+        InnerInstructions, Reward, TransactionStatusMeta, TransactionTokenBalance,
+    },
+};
+
+const MAX_TRANSACTION_STATUS_LEN: usize = 256;
+
+#[derive(Clone, Debug, ToSql)]
+#[postgres(name = "CompiledInstruction")]
+pub struct DbCompiledInstruction {
+    pub program_id_index: i16,
+    pub accounts: Vec<i16>,
+    pub data: Vec<u8>,
+}
+
+#[derive(Clone, Debug, ToSql)]
+#[postgres(name = "InnerInstructions")]
+pub struct DbInnerInstructions {
+    pub index: i16,
+    pub instructions: Vec<DbCompiledInstruction>,
+}
+
+#[derive(Clone, Debug, ToSql)]
+#[postgres(name = "TransactionTokenBalance")]
+pub struct DbTransactionTokenBalance {
+    pub account_index: i16,
+    pub mint: String,
+    pub ui_token_amount: Option<f64>,
+    pub owner: String,
+}
+
+#[derive(Clone, Debug, ToSql, PartialEq)]
+#[postgres(name = "RewardType")]
+pub enum DbRewardType {
+    Fee,
+    Rent,
+    Staking,
+    Voting,
+}
+
+#[derive(Clone, Debug, ToSql)]
+#[postgres(name = "Reward")]
+pub struct DbReward {
+    pub pubkey: String,
+    pub lamports: i64,
+    pub post_balance: i64,
+    pub reward_type: Option<DbRewardType>,
+    pub commission: Option<i16>,
+}
+
+#[derive(Clone, Debug, ToSql)]
+#[postgres(name = "TransactionStatusMeta")]
+pub struct DbTransactionStatusMeta {
+    pub status: Option<DbTransactionStatus>,
```

Comment:
I think the option here is kind of confusing with the name, because `None` means the transaction was Ok.

Maybe either:
1. Rename `status` -> `error`. Rename all `DbTransactionStatus` to `DbTransactionFailure` because they only contain failures.

or

2. Change `Option<DbTransactionStatus>` to have both success and failure options, like

```
enum `DbTransactionStatus {
      Success,
      Failure(DbTransactionFailure),
}

pub struct DbTransactionFailure {
    error_: DbTransactionError,
    error_detail: Option<String>,
}
```

Reply:
Great point! I think option 1 is more straightforward.

Reply:
Done: renamed in such way: status-> error. DbTransactionError -> DbTransactionErrorCode and DbTransactionStatus -> DbTransactionError.